src/hal/environment.cpp

```diff
@@ -129,48 +155,39 @@ void OswHal::Environment::setupStepStatistics() {
         if ((resetDay >= 0 && resetDay < 8) && (unsigned short) resetDay == currDoW + 1) // (e.g. 1 - 7 are days, 0 is disabled)
             this->_stepsSum = 0;
 
-        OSW_LOG_D("Updated steps from DoW ", lastDoW, " to DoW ", currDoW);
-        lastDoW = currDoW;
+        this->_stepsLastDoW = currDoW;
+        OSW_LOG_D("Updated steps from DoW ", this->_stepsLastDoW, " to DoW ", currDoW);
```

Comment:
Doesn't it print with the same value?

Reply:
It should, as I've just renamed the variable and moved it into the class itself. This is needed to commit the history only if needed and to prevent reading it every time from the NVS...